src/api/search.api.js

```diff
@@ -321,6 +465,22 @@ export default async function search(config) {
         )
     }
 
+    if (layersToSearch.some((layer) => layer.type === LayerTypes.KML)) {
+        allRequests.push(
+            ...layersToSearch
+                .filter((layer) => layer.type === LayerTypes.KML)
+                .map((layer) => searchKmlLayerFeatures(outputProjection, queryString, layer))
+        )
+    }
+
+    if (layersToSearch.some((layer) => layer.type === LayerTypes.GPX)) {
+        allRequests.push(
+            ...layersToSearch
+                .filter((layer) => layer.type === LayerTypes.GPX)
+                .map((layer) => searchGpxLayerFeatures(outputProjection, queryString, layer))
+        )
+    }
```

Comment:
As both `searchKmlLayerFeatures` and `searchGpxLayerFeatures` are doing a similar job (only the feature parsing differs) you could maybe try having only one single function that can deal with both types, and then join these two IFs into one